arrow/src/compute/kernels/comparison.rs

```diff
@@ -1448,7 +1448,82 @@ pub fn eq_dyn_bool_scalar(left: Arc<dyn Array>, right: bool) -> Result<BooleanAr
             eq_bool_scalar(left, right)
         }
         _ => Err(ArrowError::ComputeError(
-            "Kernel only supports BooleanArray".to_string(),
+            "eq_dyn_bool_scalar only supports BooleanArray".to_string(),
+        )),
+    };
+    result
+}
+
+/// Perform `left < right` operation on an array and a numeric scalar
+/// value. Supports BooleanArrays, and DictionaryArrays that have string values
+pub fn lt_dyn_bool_scalar(left: Arc<dyn Array>, right: bool) -> Result<BooleanArray> {
```

Comment:
I think we should update this to 

```suggestion
pub fn lt_dyn_bool_scalar(left: &dyn Array, right: bool) -> Result<BooleanArray> {
```

Rather than taking an owned `Arc` to be consistent with the work in https://github.com/apache/arrow-rs/pull/1127

I am happy to make the change as a follow on PR as well.

Reply:
since this PR is for bool i think best to update the signatures in a follow on PR.  to confirm though, this would be for all dyn scalar kernels right? scalar, utf8, and bool?

Reply:
> since this PR is for bool i think best to update the signatures in a follow on PR. to confirm though, this would be for all dyn scalar kernels right? scalar, utf8, and bool?

Yes

However, I made the change for `scalar` and `utf8` already in https://github.com/apache/arrow-rs/pull/1127 -- so how about I merge this PR and then I can make the updates to that PR as well

Reply:
sure sounds good

Reply:
Updated API in https://github.com/apache/arrow-rs/pull/1127/commits/e97f588a98efccd18e435a00a4e89841f010c94a